docs/en_US/GeneralNasInterfaces.md

```diff
@@ -83,9 +85,99 @@ Accordingly, a specified neural architecture (generated by tuning algorithm) is
+
+NNI has supported the general NAS as demonstrated above. From users' point of view, One-Shot NAS and NAS have the same search space specification, thus, they could share the same programming interface as demonstrated above, just different training modes. NNI provides four training modes:
+
+**classic_mode**: this mode is described [above](#ProgInterface), in this mode, each subgraph runs as a trial job. To use this mode, you should enable NNI annotation and specify a tuner for nas in experiment config file. [Here](https://github.com/microsoft/nni/tree/master/examples/trials/mnist-nas) is an example to show how to write trial code and the config file. And [here](https://github.com/microsoft/nni/tree/master/examples/tuners/random_nas_tuner) is a simple tuner for nas.
+
+**enas_mode**: following the training approach in [enas paper][1]. It builds the full graph based on neural architrecture search space, and only activate one subgraph that generated by the controller for each mini-batch. [Detailed Description](#ENASMode). (currently only supported on tensorflow). 
+
+To use enas_mode, you should add one more field in the `trial` config as shown below.
+```diff
+trial:
+    command: your command to run the trial
+    codeDir: the directory where the trial's code is located
+    gpuNum: the number of GPUs that one trial job needs
++   #choice: classic_mode, enas_mode, oneshot_mode
++   nasMode: enas_mode
+```
+Similar to classic_mode, in enas_mode you need to specify a tuner for nas, as it also needs to receive subgraphs from tuner (or controller using the terminology in the paper). Since this trial job needs to receive multiple subgraphs from tuner, each one for a mini-batch, two lines need to be added to the trial code to receive the next subgraph and report the report the result of the current subgraph. Below is an example:
+```python
+for _ in range(num):
+    """@nni.get_next_parameter(self.session)"""
+    loss, _ = self.session.run([loss_op, train_op])
+    """@nni.report_final_result(loss)"""
+```
+Here, `get_next_parameter` needs an arg which is the session variable that you use to train your model.
+
+**oneshot_mode**: following the training approach in [this paper][6]. Different from enas_mode which trains the full graph by training large numbers of subgraphs, in oneshot_mode the full graph is built and dropout is added to candidate inputs and also added to candidate ops' outputs. Then this full graph is trained like other DL models. [Detailed Description](#OneshotMode). (currently only supported on tensorflow). 
+
+To use oneshot_mode, you should add one more field in the `trial` config as shown below. In this mode, no need to specify tuner in the config file as it does not need tuner. (Note that you still need to specify a tuner (any tuner) in the config file in v0.9, not anymore in future releases.)
```

Comment:
"not anymore in future releases"? (Remove? because we cannot make sure now.)

Reply:
For oneshot_mode, in v0.9, user has to specify a tuner. However, specifying tuner would make this mode strange, so we will remove this constraint in the next release.

Reply:
good suggestion, let's remove it